modules/Attendance/attendance.php

```diff
@@ -16,488 +16,448 @@
+$accessNotRegistered = isActionAccessible($guid, $connection2, "/modules/Attendance/report_rollGroupsNotRegistered_byDate.php")
+    && isActionAccessible($guid, $connection2, "/modules/Attendance/report_courseClassesNotRegistered_byDate.php");
+$gibbonPersonID = ($accessNotRegistered && isset($_GET['gibbonPersonID'])) ?
+    $_GET['gibbonPersonID'] : $session->get('gibbonPersonID');
+
+// show access denied message, if needed
+if (!isActionAccessible($guid, $connection2, '/modules/Attendance/attendance.php')) {
+    $page->addError(__("You do not have access to this action."));
+    return;
 }
-else {
-	//Proceed!
-	print "<div class='trail'>" ;
-	print "<div class='trailHead'><a href='" . $_SESSION[$guid]["absoluteURL"] . "'>" . __("Home") . "</a> > <a href='" . $_SESSION[$guid]["absoluteURL"] . "/index.php?q=/modules/" . getModuleName($_GET["q"]) . "/" . getModuleEntry($_GET["q"], $connection2, $guid) . "'>" . __(getModuleName($_GET["q"])) . "</a> > </div><div class='trailEnd'>" . __('View Daily Attendance') . "</div>" ;
-	print "</div>" ;
-
-	print "<h2>" ;
-	print __("View Daily Attendance") ;
-	print "</h2>" ;
-
-	if (isset($_GET["currentDate"])==FALSE) {
-	 	$currentDate=date("Y-m-d");
-	}
-	else {
-		$currentDate=dateConvert($guid, $_GET["currentDate"]) ;
-	}
-
-	$today=date("Y-m-d");
-
-	$lastNSchoolDays = getLastNSchoolDays($guid, $connection2, $currentDate, 10, true);
-
-	$accessNotRegistered = isActionAccessible($guid, $connection2, "/modules/Attendance/report_rollGroupsNotRegistered_byDate.php") && isActionAccessible($guid, $connection2, "/modules/Attendance/report_courseClassesNotRegistered_byDate.php");
 
-	$gibbonPersonID = ($accessNotRegistered && isset($_GET['gibbonPersonID']))? $_GET['gibbonPersonID'] : $_SESSION[$guid]["gibbonPersonID"];
-
-	$form = Form::create('action', $_SESSION[$guid]['absoluteURL'].'/index.php','get');
+// define attendance filter form, if user is permit to view it
+if (isActionAccessible($guid, $connection2, '/modules/Attendance/attendance.php')) {
+    $form = Form::create('action', $session->get('absoluteURL').'/index.php', 'get');
 
+    $form->setTitle(__('View Daily Attendance'));
     $form->setFactory(DatabaseFormFactory::create($pdo));
     $form->setClass('noIntBorder fullWidth');
 
-    $form->addHiddenValue('q', "/modules/".$_SESSION[$guid]['module']."/attendance.php");
+    $form->addHiddenValue('q', "/modules/".$session->get('module')."/attendance.php");
 
-	$row = $form->addRow();
-	    $row->addLabel('currentDate', __('Date'))->description($_SESSION[$guid]['i18n']['dateFormat'])->prepend(__('Format:'));
-	    $row->addDate('currentDate')->setValue(dateConvertBack($guid, $currentDate))->isRequired();
+    $row = $form->addRow();
+    $row->addLabel('currentDate', __('Date'))->description($_SESSION[$guid]['i18n']['dateFormat'])->prepend(__('Format:'));
+    $row->addDate('currentDate')->setValue(Format::dateConvert($currentDate))->isRequired();
```

Comment:
Just tested out the branch and caught a couple more small things. The date picker UI uses the localized date, in this case just Format::date() does the trick.

```suggestion
    $row->addDate('currentDate')->setValue(Format::date($currentDate))->isRequired();
```

Reply:
Should I change all the `Format::dateConvert` to `Format::date`? Or just this one?

Reply:
Ah, good catch, if they're using `$currentDate` then yep, change then all to `Format::date`